### PR TITLE
Support `IPV6_ONLY` configurations for compute API

### PR DESCRIPTION
**Pull request by karolgorc**
related to b/360733056

This provides support to set external and internal IPV6 addresses only for VM's and templates

I'm not sure if changing `ipv6_access_config` to `Computed: true` is a breaking change but it's needed because when providing external ipv6 subnetwork the field will be filled from API. CI tests should spot this if it is breaking but would love some feedback here

**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note:enhancement
compute: `stack_type` can now be set to `IPV6_ONLY` on `google_compute_subnetwork`, `google_compute_instance`, `google_compute_instance_template` and `google_compute_region_instance_template`.
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/compute/resource_compute_instance.go.tmpl [hunk 1]
```diff
@@ -541,7 +541,8 @@ func ResourceComputeInstance() *schema.Resource {
 
 						"ipv6_access_config": {
 							Type:        schema.TypeList,
-                                                        Optional:    true,
+                            Optional:    true,
+							Computed:    true,
```
[the remark this answers is not in the document]
Comment (karolgorc) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance.go.tmpl:545]: As stated in PR's description, if the `subnetwork` given to the `network_interface` is a subnetwork of external IPv6 addresses `ipv6_access_config` will get filled from the API without user's interaction

```
resource "google_compute_subnetwork" "test" {
  name = "testing"
  network = google_compute_network.test.self_link
  stack_type = "IPV6_ONLY"
  ipv6_access_type = "EXTERNAL"
}

resource "google_compute_instance" "test" {
  name = "test"
  machine_type = "n2-standard-2"

  boot_disk {
    initialize_params {
      image = "debian-cloud/debian-11"
    }
  }

  network_interface {
    subnetwork = google_compute_subnetwork.test.self_link
    stack_type = "IPV6_ONLY"
  }
}
```

**Comment by karolgorc** [comment]
What does this fail on?

TestAccComputeInstanceNetworkIntefaceWithSecurityPolicy

Can't test this locally
```
Error: Error creating RegionSecurityPolicy: googleapi: Error 400: Invalid value for field 'resource.type': 'CLOUD_ARMOR_NETWORK'. Network Security Policies are not supported as part of the current Cloud Armor service tier., invalid
```

**Comment by slevenick** [comment]
> What does this fail on?
> 
> TestAccComputeInstanceNetworkIntefaceWithSecurityPolicy
> 
> Can't test this locally
> 
> ```
> Error: Error creating RegionSecurityPolicy: googleapi: Error 400: Invalid value for field 'resource.type': 'CLOUD_ARMOR_NETWORK'. Network Security Policies are not supported as part of the current Cloud Armor service tier., invalid
> ```

        Error: Error deleting old access_config: googleapi: Error 400: Invalid value for field 'accessConfig': 'external-nat'. Cannot delete an access config with a security policy set. Please remove the security policy first., invalid

**Comment by karolgorc** [comment]
I'm guessing that it's this subtest specifically `testAccComputeInstance_nic_securityPolicyCreateWithAccessConfigUpdateAccessConfig`

There is a weird handling of errors here on terraform's side because the error can be thrown both by the Security policy and by the Access Config (having different error messages). So if any changes occur in the API this test is very fragile.

Already fixed that one time https://github.com/GoogleCloudPlatform/magic-modules/pull/11350

Does this fail in other PR's? Or is this caused by my change?

**Review by slevenick — APPROVED** [review]

**Comment by zli82016** [comment]
@karolgorc , the test `TestAccComputeInstanceNetworkIntefaceWithSecurityPolicy` also failed during the [nightly runs](https://hashicorp.teamcity.com/test/2812118840104242594?currentProjectId=TerraformProviders_GoogleCloud_GOOGLE_BETA_NIGHTLYTESTS&expandTestHistoryChartSection=true) since this PR was merged on Nov 11. Do you have the plan to fix it soon? Or we can consider revert this PR.

**Comment by karolgorc** [comment]
Not sure why this was merged because i can't test this without Cloud Armor on my GCP environment to tell whether this was local or caused by something else =P.
> Does this fail in other PR's? Or is this caused by my change?

I have a fix. I'll create a new PR with it but i'm also looking out for any customer issues related to this as of now

**Comment by zli82016** [comment]
@karolgorc , there are not customer issues related to this now, as this change has not been released to the customers yet.

Thanks for working on the fix. The new PR needs to be merged by end of Tuesday next week, when the next release will be cut. Otherwise, this PR has to be reverted.

**Comment by zli82016** [comment]
@karolgorc , can you help me understand why adding `Computed: true` to `ipv6_access_config` causes the test failing? 

In the test `TestAccComputeInstanceNetworkIntefaceWithSecurityPolicy/access_config_update_access_config`,  [stack_type is "IPV4_IPV6"](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.tmpl#L10715), not IPV6_ONLY

Thanks.

**Comment by karolgorc** [comment]
> can you help me understand why adding Computed: true to ipv6_access_config causes the test failing?

Not sure about the API inner workings here but we have a weird race condition here where the smallest change cloud alter the error message. It's either thrown by the access config or by the security policy. At least that's what it was when last fixing it. 

Not sure why it behaves this way on "IPV6_ONLY"

We cannot really make a condition in terraform here to use `ComputedIf` because this would require queries for the type of subnetwork that's provided to the subnetwork resource itself.

Basically for now i'd say that this feature generates a diff that can only be handled by using "Computed: True" and breaks this test. Unless we want to do some weird hacks with CustomizeDiff but i'd say that would be troublesome to maintain down the line.

As i said i don't have an environment with Cloud Armor to closely look into this.
